### PR TITLE
doc: Bluetooth: Mesh: Add migration guide to v1.7.0

### PR DESCRIPTION
**Pull request by PavelVPV**
Changes noted in the migration guide:
- zephyrproject-rtos/zephyr#34606
- #5213
- #5156

Signed-off-by: Pavel Vasilyev <pavel.vasilyev@nordicsemi.no>

### CODE REVIEW

**Review by trond-snekvik — APPROVED** [review]

**Comment by PavelVPV** [comment]
I added DNM label so that PR isn't merged until https://github.com/nrfconnect/sdk-nrf/pull/5156 is merged.

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -181,6 +181,16 @@ Bluetooth mesh
   * Fixed an issue where the IV update procedure could be started immediately after the device has been provisioned.
   * Fixed multiple issues in :ref:`bt_mesh_sensor_types_readme` module.
 
+* Migration:
+
+  * The model opcode callback :c:member:`bt_mesh_model_op.func` is changed to return an error code if message processing has failed.
```
Comment (mia-ko) [inline doc/nrf/releases/release-notes-changelog.rst:186]:
```suggestion
  * The model opcode callback :c:member:`bt_mesh_model_op.func` is changed to return an error code if the message processing has failed.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -181,6 +181,16 @@ Bluetooth mesh
...
+    If you have implemented your own models, make sure to update opcode handlers of those models.
+  * :ref:`bt_mesh_scene_srv_readme` now extends :ref:`bt_mesh_dtt_srv_readme`.
+    If you use the Scene Server, make sure that the Generic Default Transition Time Server instance is present on the element that is equal or lower than the Scene Server's element.
```
Comment (mia-ko) [inline doc/nrf/releases/release-notes-changelog.rst:189]:
```suggestion
    If you are using the Scene Server, make sure that the Generic Default Transition Time Server instance is present on the element that is equal to or lower than the Scene Server's element.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -181,6 +181,16 @@ Bluetooth mesh
...
+  * :ref:`bt_mesh_light_hsl_srv_readme` and :ref:`bt_mesh_light_xyl_srv_readme` now do not instantiate :ref:`bt_mesh_lightness_srv_readme` through :c:macro:`BT_MESH_MODEL_LIGHT_HSL_SRV` and :c:macro:`BT_MESH_MODEL_LIGHT_XYL_SRV` macros accordingly.
```
Comment (mia-ko) [inline doc/nrf/releases/release-notes-changelog.rst:190]:
```suggestion
  * :ref:`bt_mesh_light_hsl_srv_readme` and :ref:`bt_mesh_light_xyl_srv_readme` no longer instantiate the :ref:`bt_mesh_lightness_srv_readme` through :c:macro:`BT_MESH_MODEL_LIGHT_HSL_SRV` and :c:macro:`BT_MESH_MODEL_LIGHT_XYL_SRV` macros respectively.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 4]
```diff
@@ -181,6 +181,16 @@ Bluetooth mesh
...
+    :c:macro:`BT_MESH_LIGHT_XYL_SRV_INIT` and :c:macro:`BT_MESH_LIGHT_HSL_SRV_INIT` macros now take pointer to :c:struct:`bt_mesh_lightness_srv` instance instead.
+    Make sure to instantiate the Light Lightness Server if you use any of those models.
```
Comment (mia-ko) [inline doc/nrf/releases/release-notes-changelog.rst:192]:
```suggestion
    Macros :c:macro:`BT_MESH_LIGHT_XYL_SRV_INIT` and :c:macro:`BT_MESH_LIGHT_HSL_SRV_INIT` now take a pointer to the :c:struct:`bt_mesh_lightness_srv` instance instead.
    Make sure to instantiate the Light Lightness Server if you are using any of these models.
```
Take a pointer or pass a pointer?

**Review by mia-ko — CHANGES_REQUESTED** [review]

**Review by mia-ko — COMMENTED** [review]
Release notes for NCS 1.7.0 PR is now open, see: https://github.com/nrfconnect/sdk-nrf/pull/5534
Our changes should go to release-notes-1.7.0.rst file instead.

File: doc/nrf/releases/release-notes-changelog.rst (hunk 4 above)
Comment (PavelVPV) [reply doc/nrf/releases/release-notes-changelog.rst:192]: User passes a pointer to the macros => macros take a pointer.

**Review by mia-ko — APPROVED** [review]

**Comment by PavelVPV** [comment]
Closing as the change is merged in https://github.com/greg-fer/fw-nrfconnect-nrf-1/pull/6.

**Pull request closed without merging**